src/LuaMultiThreaded2.cpp

```diff
@@ -68,6 +69,47 @@ extern "C" int errorHandler(LuaState * L)
 
 
 
+extern "C" static int mutexNew(LuaState * aState)
+{
+	lua_pushcfunction(aState, errorHandler);
+
+	// Push the (currently empty) mutex object to the Lua side
+	auto mutexObj = reinterpret_cast<std::mutex**>(lua_newuserdata(aState, sizeof(std::mutex**)));
+	luaL_setmetatable(aState, MUTEX_METATABLE_NAME);
+
+	// Create the new mutex:
+	*mutexObj = new std::mutex();
+	return 1;
+}
+
+
+
+
+
+
+
+/** */
+extern "C" static int mutexLock(LuaState * aState)
+{
+	auto mutexObj = reinterpret_cast<std::mutex**>(luaL_checkudata(aState, 1, MUTEX_METATABLE_NAME));
+	if (mutexObj == nullptr)
+	{
+		luaL_argerror(aState, 0, "'mutex' expected");
+		return 0;
+	}
+	(*mutexObj)->lock();
```

Comment:
`*mutexObj` may be `nullptr` at this point (if the initialization failed half-way through), so you should check that as well.